doc/interfaces/denoise.rst

```diff
@@ -0,0 +1,223 @@
+techniques such as DTI, spherical deconvolution and DKI.
+
+The basic idea behind the PCA-based denoising algorithms is to remove the components 
+of the data that are classified as noise. The Principal Components classification
+can be performed based on prior noise variance estimates or automatically based 
+on the Marcenko-Pastur distribution. In addition to noise suppression, the PCA
+algorithm can be used to get the standard deviation of the noise.
+
+We will use the ``sherbrooke_3shell`` dataset in DIPY to showcase this denoising method.
+As with any other workflow in DIPY, you can also use your own data!
+
+Create a directory where to save the denoised image (e.g.:
+``denoise_mppca_output``)::
+
+    mkdir denoise_mppca_output
+
+In order to run the MPPCA denoising method, we need to specify the location of the 
+diffusion data file, followed by the optional arguments. In this case, we will be 
+specifying the ``patch radius`` value and ``output directory``.
+
+So, we will run the command as::
+
+    dipy_denoise_mppca data/sherbrooke_3shell/HRADI193.nii.gz --patch_radius 10 --out_dir "denoise_mppca_output"
+
+This command will denoise the diffusion image and save it to the specified output directory.
+
+.. |image3| image:: https://github.com/dipy/dipy_data/blob/master/sherbrooke_3shell_original.png  
+   :scale: 70%
+   :align: middle
+.. |image4| image:: https://github.com/dipy/dipy_data/blob/master/sherbrooke_3shell_original.png
+   :scale: 70%
+   :align: middle
+
++--------------------+--------------------+
+|  Before Denoising  |  After Denoising   |
++====================+====================+
+|      |image3|      |      |image4|      |
++--------------------+--------------------+
+
+-----------------------
+Denoising using NLMEANS
+-----------------------
+
+Using the non-local means filter [Coupe08]_ and [Coupe11]_, you can denoise 3D or 4D
+images and boost the SNR of your datasets.
+
+We will use the ``cfin_multib`` dataset in DIPY to showcase this denoising method.
+As with any other workflow in DIPY, you can also use your own data!
+
+In order to run the NLMEANS denoising method, we need to specify the location of the 
+diffusion data file, followed by the optional arguments. In this case, we will be 
+specifying the ``sigma`` and ``patch radius`` values and ``output directory``.
+
+Create a directory where to save the denoised image (e.g.:
+``denoise_nlmeans_output``)::
+
+Then, we will run the command as::
+
+    dipy_denoise_nlmeans data/cfin_multib/__DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii --sigma 2 --patch_radius 2 --out_dir "denoise_nlmeans_output"
+
+The command will denoise the input diffusion volume and write the result to the specified
+output directory.
+
+.. |image5| image:: https://github.com/dipy/dipy_data/blob/master/cfin_multib_original.png  
+   :scale: 20%
+   :align: middle
+.. |image6| image:: https://github.com/dipy/dipy_data/blob/master/cfin_multib_denoise_NLMEANS.png
```

Comment:
you need to add `?raw=true` at the end of the URL.

Reply:
Addressed in 12ceeef